sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/FieldSingleDeclarationCheck.java

```diff
@@ -0,0 +1,107 @@
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>This checks ensures that classes have at most 1 field of the given className.</p>
+ *
+ * <p>This can be useful for example to ensure that only one Logger is used:</p>
+ *
+ * <pre>
+ * &lt;module name="TreeWalker"&gt;
+ *   &lt;module name="com.github.sevntu.checkstyle.checks.coding.FieldSingleDeclarationCheck"&gt;
+ *       &lt;property name="className" value="org.slf4j.Logger" /&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ *
+ * class Example {
+ *     private static final org.slf4j.Logger logger = LoggerFactory.getLogger(Example.class); // OK
+ *     private static       org.slf4j.Logger logger2; // NOK!
+ * }
+ * </pre>
+ *
+ * @author Milos Fabian, Pantheon Technologies - original author (in OpenDaylight.org)
+ * @author Michael Vorburger.ch - refactored and made more generalized for contribution to Sevntu
+ */
+public class FieldSingleDeclarationCheck extends AbstractCheck {
+
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "field.count";
+
+    /**
+     * Configuration property with class name check for.
+     */
+    private String fullyQualifiedClassName;
+
+    /**
+     * Field to remember if class was previously seen in current File.
+     */
+    private boolean hasPreviouslySeenClass;
+
+    /**
+     * Set Class of which there should only be max. 1 field per class.
+     * @param className the fully qualified name (FQN) of the class
+     */
+    public void setClassName(String className) {
+        this.fullyQualifiedClassName = className;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.VARIABLE_DEF };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        this.hasPreviouslySeenClass = false;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (fullyQualifiedClassName == null) {
+            throw new IllegalStateException("Must set mandatory className property in"
+                    + getClass().getSimpleName());
+        }
+        if (Utils.matchesFullyQualifiedName(ast, fullyQualifiedClassName)) {
+            if (hasPreviouslySeenClass) {
```

Comment:
I do not see filtering for fields.
Your Check is named - FieldSingleDeclarationCheck , but you examine fields+variables.